django/core/validators.py

```diff
@@ -249,23 +257,24 @@ def __call__(self, value):
         if not self.user_regex.match(user_part):
             raise ValidationError(self.message, code=self.code, params={"value": value})
 
+        # Only check for unsafe chars in domain part
+        if self.unsafe_chars.intersection(domain_part):
+            raise ValidationError(self.message, code=self.code, params={"value": value})
+
```

Comment:
Is this check necessary? (I think the domain regex already blocks all control characters.)

[For the benefit of other reviewers: I was going to suggest checking the entire value, not just the domain_part, for newlines etc. But I was surprised to find that older email RFCs allowed control characters the user_part, inside a quoted-string. The current RFC 5322 (from 2008) [makes that obsolete](https://datatracker.ietf.org/doc/html/rfc5322#section-4.1:~:text=Bare%20CR%2C%20bare%20LF%2C%20and%20NUL%20are%20added%20to%20obs%2Dqp%2C%0A%20%20%20obs%2Dbody%2C%20and%20obs%2Dunstruct.%20%20US%2DASCII%20control%20characters%20are%20added%20to%0A%20%20%20obs%2Dqp%2C%20obs%2Dunstruct%2C%20obs%2Dctext%2C%20and%20obs%2Dqtext.). But EmailValidator.user_regex seems to have been written against an earlier spec and specifically allows most control chars. We might want to consider dropping obsolete address support in a separate ticket, as it's unlikely to have any legitimate use today.]